README.md

```diff
@@ -145,7 +183,7 @@ code, include a script or makefile that will do the job.
 
 ## A2.5
 
-«replace this with your answer»
+sh run_program.sh (program file is statemachine.cpp, the script compiles & runs it)
```

Comment:
Nice answer, and thanks for the tests. Full marks for this.

Just a couple of observations. 

* C++ being the language it is, the command to build your code on OS X is

  ~~~ sh
  $ clang -std=c++14 -o main -l c++ *.cpp
  ~~~

* This seems to be an SMU student thing. Everyone seems to enjoy the verbosity of C++, and embrace it with code like

~~~ c++
    string tests[9];
    tests[0] = "the lazy dog ate slowly";
    tests[1] = "the smelly dog ran noisily";
    tests[2] = "the lazy smelly lazy dog ate slowly";
    tests[3] = "the smelly lazy dog ran noisily";
    tests[4] = "the dog ran slowly";
    tests[5] = "the ate slowly";
    tests[6] = "the lazy dog ate slowly noisily";
    tests[7] = "the lazy dog ate";
    tests[8] = "the lazy smelly lazy smelly smelly lazy dog ate";

    bool expected[9];
    expected[0] = true;
    expected[1] = true;
    expected[2] = true;
    expected[3] = true;
    expected[4] = true;
    expected[5] = false;
    expected[6] = false;
    expected[7] = true;
    expected[8] = true;
~~~

Why not

~~~ c++
    string tests[] = {
      "the lazy dog ate slowly",
      "the smelly dog ran noisily",
      "the lazy smelly lazy dog ate slowly",
      "the smelly lazy dog ran noisily",
      "the dog ran slowly",
      "the ate slowly",
      "the lazy dog ate slowly noisily",
      "the lazy dog ate",
      "the lazy smelly lazy smelly smelly lazy dog ate"
    };

    bool expected[] = {
      true, true, true, true, true, false, false, true, true,
    };
~~~

Data is always easier than code :)